app/models/atmosphere/appliance.rb

```diff
@@ -129,15 +127,40 @@ def owned_by?(user)
       appliance_set.user_id == user.id
     end
 
+    def prepaid_until
+      # Helper method which determines how long this appliance will remain prepaid
+      if deployments.blank?
+        # Invoking this method on an appl whith 0 deployments is a nasal demons
+        # scenario: the question doesn't make sense and neither will the answer.
+        # Returning fixed time is least likely to break the client.
+        result = Time.parse('2000-01-01 12:00')
+      else
+        result = deployments.max_by { |dep| dep.prepaid_until }.prepaid_until
+      end
+      result
+    end
+
+    # This method provided for backward compatibility
+    def billing_state
+      if deployments.any? { |d| d.billing_state == 'prepaid' }
+        'prepaid'
+      else
+        'expired'
```

Comment:
@Nuanda are you referring [this line](https://github.com/dice-cyfronet/atmosphere/blob/billing_in_deployments/app/models/atmosphere/deployment.rb#L22)? If yes then my `prepaid?` method is still valid. According to `prepaid` and `expired` in `billing_state` you are absolutely right - I was not checking how it is used - my fault.